python/lsst/daf/ingest/ingestCatalogTask.py

```diff
@@ -0,0 +1,580 @@
+    ArrayI = FieldFormatter(lambda f: _sqlTypeForArray("i", f), lambda v: _formatArray("i", v)),
+    ArrayF = FieldFormatter(lambda f: _sqlTypeForArray("f", f), lambda v: _formatArray("f", v)),
+    ArrayD = FieldFormatter(lambda f: _sqlTypeForArray("d", f), lambda v: _formatArray("d", v)),
+)
+
+
+def canonicalizeFieldName(fieldName):
+    """Return a MySQL-compatible version of the schema field name, for now by
+    changing any non-word characters to underscores.
+    """
+    return re.sub(r"[^\w]", "_", fieldName)
+
+def quoteIdentifier(identifier):
+    """Return a MySQL compatible version of the given identifier by quoting
+    it with back-ticks (and escaping embedded back-ticks).
+    """
+    return "`" + identifier.replace("`", "``") + "`"
+
+def aliasesFor(name, mappings):
+    """Compute and return the set of possible aliases for the given field name.
+
+    The afw table library processes a given name F by replacing the longest
+    prefix of F that can be found in a schema's alias map with the corresponding
+    alias target. Replacement is repeated until either no such prefix is found,
+    or N replacements have been made (where N is the size of the alias map).
+
+    The mappings argument must be a sorted list of substitutions, where a single
+    substitution is a 2-tuple of strings (source, target).
+    """
+    n = 0
+    aliases, names = set(), set()
+    names.add(name)
+    while n < len(mappings) and len(names) > 0:
+        # Perform one round of reverse alias substitution
+        n += 1
+        newNames = set()
+        for name in names:
+            for i, (source, target) in enumerate(mappings):
+                if name.startswith(target):
+                    alias = source + name[len(target):]
+                    # alias is only valid if source is its longest prefix in mappings.
+                    # A prefix strictly longer than source must occur after it in the
+                    # sorted list of mappings.
+                    valid = True
+                    for s, _ in mappings[i + 1:]:
+                        if not s.startswith(source):
+                            break;
+                        if alias.startswith(s):
+                            valid = False
+                            break
+                    if valid:
+                        aliases.add(alias)
+                        newNames.add(alias)
+        names = newNames
+    return aliases
+
+
+class IngestCatalogConfig(pexConfig.Config):
+    """Configuration for the IngestCatalogTask.
+    """
+
+    allowReplace = pexConfig.Field(
+        "Allow replacement of existing rows with the same unique IDs",
+        bool, default=False
+    )
+
+    maxQueryLen = pexConfig.Field(
+        "Maximum length of a query string. None means use a non-standard, "
+        "database-specific way to get the maximum.",
+        int, optional=True, default=None
+    )
```

Comment:
Similar to the comment below (which I wrote first...) -- is there value in having this be configurable? Under what circumstances would the user want the query length to be smaller than the maximum supported by the database? If we were supporting multiple database back-ends and they didn't all provide ways to determine this in code I can see the value of the configuration parameter, but the assumption of MySQL is fairly baked in elsewhere.